MatrixSDK/Background/MXBackgroundSyncMemoryStore.swift

```diff
@@ -0,0 +1,127 @@
+// 
+// Copyright 2020 The Matrix.org Foundation C.I.C
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+
+//  error domain
+let MXBackgroundSyncMemoryStoreErrorDomain: String = "MXBackgroundSyncMemoryStoreErrorDomain"
+
+//  error codes
+enum MXBackgroundSyncMemoryStoreErrorCode: Int {
+    case userIDMissing = 1001   // User ID is missing in credentials
+}
+
+/// Fake memory store implementation. Uses some real values from an MXFileStore instance.
+class MXBackgroundSyncMemoryStore: MXMemoryStore {
```

Comment:
It is not really an in-memory store like the name can suppose.
Why not calling it `MXBackgroundMXStore` to indicate this is another implementation of MXStore. Or `MXBackgroundStore`

Reply:
Picked `MXBackgroundStore`